pkgs/development/interpreters/python/cpython/default.nix

```diff
@@ -302,9 +302,12 @@ in with passthru; stdenv.mkDerivation {
     ./3.8/0001-On-all-posix-systems-not-just-Darwin-set-LDSHARED-if.patch
     # Use sysconfigdata to find headers. Fixes cross-compilation of extension modules.
     ./3.7/fix-finding-headers-when-cross-compiling.patch
-  ] ++ optionals stdenv.hostPlatform.isLoongArch64 [
+  ] ++ optionals ((stdenv.hostPlatform.isLoongArch64 || stdenv.buildPlatform != stdenv.hostPlatform) && pythonOlder "3.12") [
     # https://github.com/python/cpython/issues/90656
     ./loongarch-support.patch
+  ] ++ optionals (stdenv.buildPlatform != stdenv.hostPlatform && pythonAtLeast "3.11" && pythonOlder "3.13") [
+    # backport https://github.com/python/cpython/issues/95855
+    ./platform-triplet-detection.patch
```

Comment:
Would it not make more sense to apply the patch unconditionally, so we're minimizing differences between cross and not?

Reply:
Yes, _however_ that is a mass rebuild, which I guess we would want to avoid even on staging-next at this stage

Reply:
I can make a second PR that removes this guard targeting staging